google-cloud-bigquery/src/test/java/com/google/cloud/bigquery/it/ITBigQueryTest.java

```diff
@@ -4142,6 +4142,61 @@ public void testUnnestRepeatedRecordNamedQueryParameter() throws InterruptedExce
     }
   }
 
+  @Test
+  public void testUnnestRepeatedRecordNamedQueryParameterFromDataset() throws InterruptedException {
+    List<QueryParameterValue> tuples = new ArrayList<>();
+      QueryParameterValue statusValue = QueryParameterValue.string("single");
+    QueryParameterValue addressValue = QueryParameterValue.string("123 this lane");
+    QueryParameterValue cityValue = QueryParameterValue.string("Torono");
+    QueryParameterValue stateValue = QueryParameterValue.string("ON");
+        QueryParameterValue zipValue = QueryParameterValue.string("1h2j34");
+    QueryParameterValue numberOfYearsValue = QueryParameterValue.string("3");
+
+      Map<String, QueryParameterValue> struct = new HashMap<>();
```

Comment:
The order of fields inside structs matters when doing comparisons, and `HashMap` does not maintain insertion order. Because of these reasons, the `IN UNNEST(addresses)` query doesn't return any results.

I tried replacing `HashMap` with a `LinkedHashMap` and using `ImmutableMap.of`, both options made the `IN UNNEST(addresses)` query work as intended and return the correct row.